apps/meteor/app/slackbridge/server/RocketAdapter.js

```diff
@@ -134,22 +134,23 @@ export default class RocketAdapter {
 			try {
 				if (!slack.getSlackChannel(rocketMessage.rid)) {
 					// This is on a channel that the rocket bot is not subscribed
-					return;
+					continue;
 				}
 				rocketLogger.debug('onRocketMessage', rocketMessage);
 
 				if (rocketMessage.editedAt) {
 					// This is an Edit Event
 					await this.processMessageChanged(rocketMessage, slack);
-					return rocketMessage;
+					continue;
 				}
 				// Ignore messages originating from Slack
 				if (rocketMessage._id.indexOf('slack-') === 0) {
-					return rocketMessage;
+					continue;
 				}
 
 				if (rocketMessage.file) {
-					return this.processFileShare(rocketMessage, slack);
+					await this.processFileShare(rocketMessage, slack);
```

Comment:
This function call has a side effect since it changes the content of `rocketMessage.msg` [here](https://github.com/RocketChat/Rocket.Chat/blob/c2cf2d773adb161f331bd3736a1a94e5b21dd035/apps/meteor/app/slackbridge/server/RocketAdapter.js#L211) :grimacing: this could create issues for all next adapters
I believe we want to use the same `rocketMessage` object that is provided to this method for **all** adapters (in case a message needs to be handled by multiple ones)

Reply:
I've done a tiny refactor in `processFileShare()` that should do the trick to avoid this edge case from happening.

Reply:
Not sure I'm confident about this refactor. Shouldn't we return the message with the side effects generated by this function?
With your refactor, we're currently returning the same `rocketMessage` that is provided as a param to this function :thinking:

Reply:
> Not sure I'm confident about this refactor. Shouldn't we return the message with the side effects generated by this function? With your refactor, we're currently returning the same `rocketMessage` that is provided as a param to this function 🤔

Not sure I follow, @matheusbsilva137 , with the refactor we are not replacing the original content of `rocketMessage` anymore.

> If all adapters will end up doing the same to the message, why not doing the side effect before looping it?
> 
> So the func has 0 side effects and it's just in charge of sending the message to slack.

I really like this ideia, tho. It would avoid processing the same thing over and over again.

Reply:
> with the refactor we are not replacing the original content of `rocketMessage` anymore.

I was thinking: shouldn't we? Because we currently change it, but only **once**
Since this methods is used in a callback, I thought the next ones might need it. But I've just checked again the change we apply to `msg` and I don't feel like we were doing it on purpose. I'm fine with your refactor, then :)